tests/h/presenters/annotation_json_test.py

```diff
@@ -1,208 +1,116 @@
-        document_asdict.return_value = {}
 
-        presented = AnnotationJSONPresenter(context).asdict()
-        assert presented["id"] == "the-real-id"
+        DocumentJSONPresenter.assert_called_once_with(annotation.document)
+        DocumentJSONPresenter.return_value.asdict.assert_called_once_with()
 
-    def test_asdict_extra_uses_copy_of_extra(
-        self, document_asdict, groupfinder_service, links_service
-    ):
-        extra = {"foo": "bar"}
-        ann = mock.Mock(id="my-id", extra=extra)
-        context = AnnotationContext(ann, groupfinder_service, links_service)
-        document_asdict.return_value = {}
+    def test_asdict_extra_inherits_correctly(self, annotation, context):
+        annotation.extra = {"id": "DIFFERENT"}
 
-        AnnotationJSONPresenter(context).asdict()
+        presented = AnnotationJSONPresenter(context).asdict()
 
-        # Presenting the annotation shouldn't change the "extra" dict.
-        assert extra == {"foo": "bar"}
+        # We can't override things (we are applied first)
+        assert presented["id"] == annotation.id
+        # And we aren't mutated
+        assert annotation.extra == {"id": "DIFFERENT"}
 
-    def test_asdict_merges_formatters(
-        self, groupfinder_service, links_service, get_formatter
-    ):
-        ann = mock.Mock(id="the-real-id", extra={})
-        context = AnnotationContext(ann, groupfinder_service, links_service)
-        presenter = AnnotationJSONPresenter(
-            context,
-            formatters=[
-                get_formatter({"flagged": "nope"}),
-                get_formatter({"nipsa": "maybe"}),
-            ],
-        )
+    def test_asdict_merges_formatters(self, annotation, context, get_formatter):
+        formatters = [
+            get_formatter({"flagged": "nope"}),
+            get_formatter({"nipsa": "maybe"}),
+        ]
 
-        presented = presenter.asdict()
+        presented = AnnotationJSONPresenter(context, formatters=formatters).asdict()
 
         assert presented["flagged"] == "nope"
         assert presented["nipsa"] == "maybe"
 
-    def test_immutable_formatters(
-        self, groupfinder_service, links_service, get_formatter
-    ):
-        """Double-check we can't mutate the formatters list after the fact.
-
-        This is an extra check just to make sure we can't accidentally change
-        the constructor so that it simply aliases the list that's passed in,
-        leaving us open to all kinds of mutability horrors.
-
-        """
-        ann = mock.Mock(id="the-real-id", extra={})
-        context = AnnotationContext(ann, groupfinder_service, links_service)
-        formatters = []
-
-        presenter = AnnotationJSONPresenter(context, formatters)
-        formatters.append(get_formatter({"enterprise": "synergy"}))
-        presented = presenter.asdict()
-
-        assert "enterprise" not in presented
+        for formatter in formatters:
+            formatter.format.assert_called_once_with(context)
```

Comment:
Do we also want to somehow assert that the formatters changes were applied?

Reply:
There's a separate test that asserts that the formatters results are merged in. This test's focus is exclusively on the mutated one _not_ being included.